ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/AverageIndicator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2024 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.helpers;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.CachedIndicator;
+import org.ta4j.core.num.Num;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Indicator to calculate the average of a list of other indicators.
+ */
+public class AverageIndicator extends CachedIndicator<Num> {
+    private final List<Indicator<Num>> indicators;
+
+    public AverageIndicator(Indicator<Num>... indicators) {
+        super(indicators[0]);
```

Comment:
Please perform some input validation, at minimum the null and empty list cases.

Reply:
How should this look like? `super()` and `this()` must be first statement in a constructor. This implementation would anyway through an `ArrayIndexOutOfBoundsException`.

Either way, I implemented another constructor accepting a list of indicators and calling `super()` with `getFirst()`. But the situation is the same, just in this other case it will through a `NoSuchElementException`.

A manual validation would do the same, just the runtime exception would be a different one.
```
if (indicators == null || indicators.isEmpty()) {
    throw new IllegalArgumentException("...");
}
```